--- a/tricore.ldefs
+++ b/tricore.ldefs
@@ -54,4 +54,40 @@
     <compiler name="default" spec="tricore.cspec" id="default"/>
     <external_name tool="DWARF.register.mapping.file" name="tricore.dwarf"/>
   </language>
+  <language processor="tricore"
+            endian="little"
+            size="32"
+            variant="TC179x"
+            version="1.3"
+            slafile="tricore.sla"
+            processorspec="tc179x.pspec"
+	    manualindexfile="../manuals/tricore.idx"
+            id="tricore:LE:32:tc179x">
+    <description>Infineon Tricore Embedded Processor TC179x</description>
+    <compiler name="default" spec="tricore.cspec" id="default"/>
+    <external_name tool="DWARF.register.mapping.file" name="tricore.dwarf"/>
+  </language>
+  <language processor="tricore"
+            endian="little"
+            size="32"
+            variant="TC1791_384"
+            version="1.3"
+            slafile="tricore.sla"
+            processorspec="tc1791_384.pspec"
+	    manualindexfile="../manuals/tricore.idx"
+            id="tricore:LE:32:tc1791_384">
+    <description>Infineon Tricore Embedded Processor TC1791_384</description>
+    <compiler name="default" spec="tricore.cspec" id="default"/>
+    <external_name tool="DWARF.register.mapping.file" name="tricore.dwarf"/>
+  </language>
+  <language processor="tricore"
+            endian="little"
+            size="32"
+            variant="TC1798"
+            version="1.3"
+            slafile="tricore.sla"
+            processorspec="tc1798.pspec"
+	    manualindexfile="../manuals/tricore.idx"
+            id="tricore:LE:32:tc1798">
+    <description>Infineon Tricore Embedded Processor TC1798</description>
+    <compiler name="default" spec="tricore.cspec" id="default"/>
+    <external_name tool="DWARF.register.mapping.file" name="tricore.dwarf"/>
+  </language>
  </language_definitions>
